src/main/java/duke/tasks/TaskManager.java

```diff
@@ -0,0 +1,133 @@
+package duke.tasks;
+
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.typeadapters.RuntimeTypeAdapterFactory;
+import duke.utils.PersistentList;
+import duke.utils.ResourceHandler;
+import duke.utils.Schedulable;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Manages {@code Task} objects.
+ */
+public class TaskManager {
+    /** List of {@code Task} objects. */
+    private final List<Task> tasks = new PersistentList<>("./data/tasks.txt",
+            new TypeToken<ArrayList<Task>>(){}.getType(),
+            RuntimeTypeAdapterFactory.of(Task.class)
+                    .registerSubtype(Deadline.class)
+                    .registerSubtype(Event.class)
+                    .registerSubtype(ToDo.class));
+
+    /** Constructs a {@code TaskManager} object. */
+    public TaskManager() {}
+
+    /**
+     * Adds a new {@code Task} to the {@code TaskManager}.
+     *
+     * @param task the {@code Task} object to be added.
+     * @return a string representation of the action of adding a {@code Task}.
+     */
+    public String addTask(Task task) {
+        tasks.add(task);
+        String taskCountKey = "taskManager.taskCount." + (tasks.size() == 1 ? "singular" : "plural");
+        String template = String.format("%s\n%s", ResourceHandler.getString("taskManager.addTask"),
+                ResourceHandler.getString(taskCountKey));
+        return MessageFormat.format(template, task, tasks.size());
+    }
+
+    /**
+     * Removes a {@code Task} from the {@code TaskManager}.
+     *
+     * @param listIndex the index of the {@code Task} in the {@code TaskManager} list.
+     * @return a string representation of the action of removing a {@code Task}.
+     */
+    public String removeTask(int listIndex) {
+        Task removedTask = tasks.remove(listIndex);
+        String taskCountKey = "taskManager.taskCount." + (tasks.size() == 1 ? "singular" : "plural");
+        String template = String.format("%s\n%s", ResourceHandler.getString("taskManager.removeTask"),
+                ResourceHandler.getString(taskCountKey));
+        return MessageFormat.format(template, removedTask, tasks.size());
+    }
+
+    /**
+     * Mark a {@code Task} as done.
```

Comment:
Consider changing Mark to Marks to follow Java coding Standard.

Reply:
Thanks for catching this!